### PR TITLE
#2683: getFFINodes

### LINKED ISSUE #2683: BinarizeParseMojo.java:148-154: This method may be more...
The puzzle `2649-2559f1c7` from #2649 has to be resolved: 

https://github.com/objectionary/eo/blob/14ca213bd98ebb45341b93cb34ba094dc371e6e8/eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java#L148-L154

The puzzle was created by @rultor on 07-Dec-23. 

Estimate: 90 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes #2683 

## PR-Codex overview
This PR focuses on refactoring the `BinarizeParseMojo` class to improve the extensibility of adding new FFI inserts. 

### Detailed summary
- Replaced the `List` import with `Map` and `BiFunction` imports
- Added a `FACTORY` map that matches FFI insert xpath to building of `FFINode`
- Refactored the `getFFIs` method to use the `FACTORY` map for creating FFI nodes

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@maxonfjvipon please take a look

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java [hunk 1]
```diff
@@ -112,6 +114,20 @@ public final class BinarizeParseMojo extends SafeMojo {
      */
     private Names names;
 
+    /**
+     * Map that matches ffi insert xpath to building of FFINode.
+     */
+    private final Map<String, Function<XML, FFINode>> factory = new MapOf<>(
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java:120]: @levBagryansky can we do it static?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java [hunk 2]
```diff
@@ -145,29 +161,18 @@ private XML addFFIs(
...
-                    this.names,
-                    this.targetDir.toPath().resolve("Lib"),
-                    this.eoPortalDir.toPath(),
-                    this.generatedDir.toPath().resolve("EOrust").resolve("natives")
-                )
-            );
-        }
+        final Collection<FFINode> ret = new ArrayList<>(0);
+        final XML injected = this.addFFIs(input);
+        this.factory.forEach(
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java:169]: @levBagryansky let's use just simple `for` here, or `forEach` inside lambda in order to keep the code noncontradictory

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@levBagryansky good one, just a few comments from my side

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java (hunk 1 above)
Comment (levBagryansky) [reply eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java:120]: @maxonfjvipon no since we use `this` in definition

**Review by maxonfjvipon — APPROVED** [review]
@levBagryansky LGTM! Thanks

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**